t/import_csv_file.t

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/perl -w
+
+use Modern::Perl '2017';
+
+use Getopt::Long qw/GetOptions/;
+use Log::Any::Adapter 'TAP';
+use Mock::Quick qw/qobj qmeth/;
+use Test::MockModule;
+use Test::More;
+
+use Data::DeepAccess qw(deep_exists deep_get deep_set);
+use File::Basename "dirname";
+use File::Path qw/make_path remove_tree/;
+
+use ProductOpener::Config '$data_root';
+use ProductOpener::Data qw/execute_query get_products_collection/;
+use ProductOpener::Producers qw/load_csv_or_excel_file convert_file/;
+use ProductOpener::Products "retrieve_product";
+use ProductOpener::Store "store";
+use ProductOpener::Test qw/:all/;
+
+
+my $test_id = "import_csv_file";
+my $test_dir = dirname(__FILE__);
+my $inputs_dir = "$test_dir/inputs/$test_id/";
+my $expected_dir = "$test_dir/expected_test_results/$test_id/";
+my $outputs_dir = "$test_dir/outputs/$test_id";
+make_path($outputs_dir);
+
+my $usage = <<TXT
+
+The expected results of the tests are saved in $test_dir/expected_test_results/$test_id
+
+To verify differences and update the expected test results,
+actual test results can be saved by passing --update-expected-results
+
+The directory will be created if it does not already exist.
+
+TXT
+;
+
+
+my $update_expected_results;
+
+GetOptions ("update-expected-results"   => \$update_expected_results)
+  or die("Error in command line arguments.\n\n" . $usage);
+
+# fake image download using input directory instead of distant server
+sub fake_download_image ($) {
+    my $image_url = shift;
+
+    my $fname = (split(m|/|, $image_url))[-1];
+    my $image_path = $inputs_dir . $fname;
+    my $response = qobj(
+        is_success => qmeth { return (-e $fname); },
+        decoded_content => qmeth {
+            open(my $image, "<r", $fname);
+            my $content = <$image>;
+            close $image;
+            return $content;
+        },
+    );
+    return $response;
+}
+
+
+# fields we don't want to check for they vary from test to test
+my @fields_ignore_content = qw(last_modified_t created_t owner_fields sources.0.import_t);
```

Comment:
```suggestion
my @fields_ignore_content = qw(last_modified_t created_t owner_fields sources.0.import_t entry_dates_tags last_edit_dates_tags);
```

Otherwise the tests won't work the next day:

```
    "entry_dates_tags" : [
-      "2022-04-04",
+      "2022-04-06",
       "2022-04",
       "2022"
    ],
@@ -364,7 +365,7 @@
       "en:1"
    ],
    "last_edit_dates_tags" : [
-      "2022-04-04",
+      "2022-04-06",
       "2022-04",
       "2022"
    ],

```